rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -225,4 +228,48 @@ public static boolean isValidJSONObject(String str) {
             return false;
         }
     }
+
+    public static void verifyUserAuthorization() {
+        UserVerifyAuthTask.getInstance().run();
+    }
+
+    private static class UserVerifyAuthTask {
+
+        private static UserVerifyAuthTask INSTANCE;
+        private boolean isFinish = true;
+
+        public static UserVerifyAuthTask getInstance() {
+            if (INSTANCE == null) {
+                INSTANCE = new UserVerifyAuthTask();
+            }
+            return INSTANCE;
+        }
+
+        public void run() {
+            if (isFinish) {
+                isFinish = false;
+                new AsyncTask<Void, Void, Void>() {
```

Comment:
we don't create the asynctask afresh with the singleton approach, it should be accessed via a call to `getInstance`

Reply:
https://stackoverflow.com/questions/6879584/how-to-run-the-same-asynctask-more-than-once